cleanup pir analysis passes

o- opened the pull request:
addresses review for #80, thanks @mhyee

o- committed:
cleanup pir analysis passes

addresses review for #80, thanks @mhyee

o- commented:
ping

mhyee commented:
Sorry, I don't have the bandwidth or a good enough understanding of the optimizer to comment. At least, not anytime soon.

o- commented:
sure I understand. It is almost exclusively in response to your comments. so I will count this as addressing review and merge it, since nobody else seems to have further comments
